### PR TITLE
829 - details form validation

### PR DESCRIPTION
**Pull request by Cooryd**

### CODE REVIEW

File: src/initial-submission/components/DetailsForm.test.tsx [hunk 1]
```diff
@@ -357,4 +359,108 @@ describe('DetailsForm', (): void => {
...
+        };
+
+        it('shows error message if title is empty', async () => {
+            const { getByText } = renderDetails();
+
+            fireEvent.click(getByText('TEST BUTTON'));
+            await waitFor(() => expect(getByText('details.validation.title-required')).toBeInTheDocument());
+        });
+
+        it('shows error message if subjects is empty', async () => {
```
Comment (Cooryd) [inline src/initial-submission/components/DetailsForm.test.tsx:397]: We should probably add a test case for the `feature` article type not requiring a min set of values

Comment (diversemix) [reply src/initial-submission/components/DetailsForm.test.tsx:397]: How would that work with "subject-min" ... does this mean the message needs to be dynamic?

Comment (diversemix) [reply src/initial-submission/components/DetailsForm.test.tsx:397]: Oh! Just read the code below... yes probably a good idea to test that branch in validation :)

File: src/initial-submission/components/DetailsForm.tsx [hunk 2]
```diff
@@ -32,23 +33,57 @@ const defaultManuscriptDetails = (values: ManuscriptDetails): ManuscriptDetails
...
-    //             .of(yup.string())
-    //             .min(1, t('details.validation.subjects-min'))
-    //             .max(2, t('details.validation.subjects-max'))
-    //             .required(t('details.validation.subjects-required')),
-    //     }),
-    // });
+    const schema = yup.object().shape({
+        title: yup.string().required(t('details.validation.title-required')),
+        subjects: yup.array().when('articleType', {
+            is: (articleType: string) => articleType && articleType === 'feature',
```
Comment (diversemix) [inline src/initial-submission/components/DetailsForm.tsx:39]: As-per test comment

File: src/initial-submission/components/DetailsForm.test.tsx (hunk 1 above)
Comment (Cooryd) [reply src/initial-submission/components/DetailsForm.test.tsx:397]: Add the feature test case 👍

**Pull request merged**